### PR TITLE
Update "Simple Model Comparison" notebook

### LINKED ISSUE #602: [DOCS] Simple Model Comparison example improvement
Currently, the Simple Model Comparison notebook lacks some explanations.
Please follow the guidelines from this issue: #543 to improve it

### PR DESCRIPTION
**Pull request by matanper**
resolve #602

### CODE REVIEW

File: docs/source/examples/checks/performance/simple_model_comparison.ipynb [hunk 1]
```diff
@@ -5,118 +5,87 @@
    "id": "840ea662",
```
Comment (shir22) [inline docs/source/examples/checks/performance/simple_model_comparison.ipynb:5]: <p>Title:</p><p>What is the purpose of the simple model check?</p><p> </p><p>and add another sentence about the "motivation" to compare to a baseline.... (basically - your two first sentences are great, would just elabaorate them a bit more... make them ~3)</p><p> </p><p>The second paragraph - its not very clear what is the default behavior, and what/how can the user choose...</p><p>For example after reading I don't understand if what the check does is "runs the three options", then chooses the highest scoring ones, or alternatively it "looks at my data, label and model type and heuristically chooses which model to train"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchecks/deepchecks/pull/613/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (shir22) [inline docs/source/examples/checks/performance/simple_model_comparison.ipynb:5]: <p>typo - overriden</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchecks/deepchecks/pull/613/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

File: docs/source/examples/checks/performance/simple_model_comparison.ipynb [hunk 2]
```diff
@@ -5,118 +5,84 @@
...
```
Comment (shir22) [inline docs/source/examples/checks/performance/simple_model_comparison.ipynb:5]: <p>"If the user model achieves less or similar performance to the simple model, than there is probably an issue with the model (for example, the model may not be treating class imbalance properly). " -&gt;</p><p>"If the user model achieves less or a similar score to the simple model, this is an indicator for a possible problem with the model (e.g. it wasn't trained properly)"</p><p> </p><p>"There are 3 types of simple models exists in the check. " -&gt; grammar inconsistency</p><p>"Contstant" -&gt; "Constant"</p><p> </p><p>"&nbsp;In regression is mean value, in classification the most common value." -&gt; grammar (e.g. add "is equal to the...")</p><p>Random - select by random according to which distribution?</p><p>"Runs" -&gt; trains a simple decision tree model...?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchecks/deepchecks/pull/613/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (shir22) [inline docs/source/examples/checks/performance/simple_model_comparison.ipynb:5]: <p>second paragraph, change to:</p><p> </p><p>Note the we are demonstrating on a classification task, but the check also works for regression tasks. For classificaion we will see the metrics per class, and for regression we'll have a single result per metric</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchecks/deepchecks/pull/613/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

Comment (shir22) [inline docs/source/examples/checks/performance/simple_model_comparison.ipynb:5]: <p>Do you think it is helpful to demonstrate all three?</p><p>(why not demonstrate just one?)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchecks/deepchecks/pull/613/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (shir22) [inline docs/source/examples/checks/performance/simple_model_comparison.ipynb:5]: <p>"We can see in the results that the check calculates score on for each class in the dataset, and compare the score between our model and the simple model. " -&gt;</p><p>We can see in the results that the check calculates the score for each class in the dataset, and compares the scores between our model and the simple model.</p><p> </p><p>"the check also returns a value which contains all the information which is needed to be used for validation (using conditions)." -&gt;</p><p>"the check also returns a value which includes all of the information that is needed for defining the conditions for validation."</p><p> </p><p> </p><ul><li>scores - for each metric and class returns the numeric score</li><li>classes - the classes that exist in the data</li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchecks/deepchecks/pull/613/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Comment (shir22) [inline docs/source/examples/checks/performance/simple_model_comparison.ipynb:5]: <p>In classification -&gt; For classification</p><p>separately</p><p>"and if a single class doesn't pass the" -&gt; 'and if there is a class that doesn't pass the"</p><p> </p><p>Let's add a condition to the check and see what happens when it fails:</p><p> </p><p> </p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchecks/deepchecks/pull/613/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

**Review by shir22 — CHANGES_REQUESTED** [review]
btw - just a thought - why don't we try all of the simple models and choose the best performing one? (I mean - that should be the default IMO. and if we want we can let the user override. because the idea of the check is "is there a naive baseline that performs as well as me...")

File: docs/source/examples/checks/performance/simple_model_comparison.ipynb (hunk 2 above)
Comment (matanper) [reply docs/source/examples/checks/performance/simple_model_comparison.ipynb:5]: <p>umm it just how it was before, but you are right</p>

File: docs/source/examples/checks/performance/simple_model_comparison.ipynb [hunk 3]
```diff
@@ -5,118 +5,76 @@
...
```
Comment (shir22) [inline docs/source/examples/checks/performance/simple_model_comparison.ipynb:5]: <p>"There are 3 types of simple models exists in the check. " -&gt; grammar still incorrect (did you change it?)</p><p>ok nvm lets just rewrite this:</p><p>The check has three possible "simple model" heuristics, from which one is chosen and compared to. By default the check uses the "Constant" heuristic, which can be overriden in the checks' parameters. There is no simple model which is more "correct" to use, each gives a different baseline to compare to, and you may experiment with the different types and see how it performs on your data.</p><p> </p><p>given that the params are without capslock, probably better also here... </p><p>phrasing suggestion (fixing some grammar also):</p><p>A <strong>constant</strong> model - the default. For regression the prediction is ....</p><p>A <strong>random</strong> model - draws the prediction from the distribution of the labels in the train.</p><p>A <strong>tree</strong> model - trains a simple decision tree with a given depth.</p><p> </p><p>(how can the tree depth be defined?)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchecks/deepchecks/pull/613/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (shir22) [inline docs/source/examples/checks/performance/simple_model_comparison.ipynb:5]: <pre class="ql-syntax" spellcheck="false">Line #4.    check = SimpleModelComparison(simple_model_type='constant')
</pre><p>Given that this is the default, lets do 'tree'? its more interesting... and also demonstrate how to affect tree depth? (if this happens from here...?)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchecks/deepchecks/pull/613/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (shir22) [inline docs/source/examples/checks/performance/simple_model_comparison.ipynb:5]: <p>Title -</p><p>Observe the Check's Output</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchecks/deepchecks/pull/613/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

File: docs/source/examples/checks/performance/simple_model_comparison.ipynb [hunk 4]
```diff
@@ -5,141 +5,81 @@
...
```
Comment (shir22) [inline docs/source/examples/checks/performance/simple_model_comparison.ipynb:5]: <p>Note that we are demonstrating on a classification task, but the check also works for regression tasks. For classificaion we will see the metrics per class, and for regression we'll have a single result per metric.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/deepchecks/deepchecks/pull/613/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

File: docs/source/examples/checks/performance/simple_model_comparison.ipynb [hunk 5]
```diff
@@ -67,7 +67,7 @@
     "\n",
     "We will run the check with the **tree** model type. The check will use the default metric defined in deepchecks for classification. This can be overridden by providing an alternative scorer using the `alternative_scorers` parameter.\n",
     "\n",
-    "Note the we are demonstrating on a classification task, but the check also works for regression tasks. For classificaion we will see the metrics per class, and for regression we'll have a single result per metric."
+    "Note the we are demonstrating on a classification task, but the check also works for regression tasks. For classification we will see the metrics per class, and for regression we'll have a single result per metric."
```
Comment (shir22) [inline docs/source/examples/checks/performance/simple_model_comparison.ipynb:70]:
```suggestion
    "Note that we are demonstrating on a classification task, but the check also works for regression tasks. For classification we will see the metrics per class, and for regression we'll have a single result per metric."
```

Comment (shir22) [reply docs/source/examples/checks/performance/simple_model_comparison.ipynb:70]: didn't let me commit this suggestion :-|

Comment (matanper) [reply docs/source/examples/checks/performance/simple_model_comparison.ipynb:70]: fixed

**Review by shir22 — APPROVED** [review]

**Pull request merged**